database/gormdb.go

```diff
@@ -375,7 +376,31 @@ func (db *GormDB) CreateSubmission(submission *pb.Submission) error {
 	if assignment+group != 2 {
 		return gorm.ErrRecordNotFound
 	}
-	return db.conn.Create(submission).Error
+
+	// If a submission for the given assignment and student/group already exists, update it.
+	// Otherwise create a new submission record
+	query := &pb.Submission{
+		AssignmentID: submission.GetAssignmentID(),
+		UserID:       submission.GetUserID(),
+		GroupID:      submission.GetGroupID(),
+	}
+
+	// We want the last record as there can be multiple submissions
+	// for the same submitter and lab in the legacy database.
+	if err := db.conn.Last(query, query).Error; err != nil && err != gorm.ErrRecordNotFound {
+		return err
+	}
+
+	return db.conn.Where(query).Assign(submission).FirstOrCreate(query).Error
+	/*
```

Comment:
We should avoid committing commented code.

Reply:
Done.